lib/figaro.rb

```diff
@@ -0,0 +1,65 @@
+require 'yaml'
+
+class Figaro
+  class << self
+    attr_reader :env
+  end
+
+  def self.setup(path, env = Rails.env)
+    @env ||= Environment.new(path, env)
+  end
+
+  def self.require_keys(keys)
+    env.require_keys(keys)
+  end
+
+  class Environment
+    attr_reader :config
+
+    def initialize(configuration, env)
+      @config = {}
+
+      keys = Set.new(configuration.keys - %w[production development test])
+      keys |= configuration[env]&.keys || []
+
+      keys.each do |key|
+        value = configuration.dig(env, key) || configuration[key]
+        env_value = ENV[key]
+
+        if env_value
+          warn "WARNING: #{key} is being loaded from ENV instead of application.yml"
+          @config[key] = env_value
+        else
+          @config[key] = value
+          ENV[key] = value if key == key.upcase
```

Comment:
I don't think we want this `key == key.upcase` here, our gems like lexisnexis read the lowercase versions:

https://github.com/18F/identity-lexisnexis-api-client-gem/blob/a75487c2377bb7d9edb9a24411f308f782a872f2/lib/lexisnexis/request.rb#L42

Reply:
Ah, didn't realize those depended on it, will do it for all keys